client/src/app/pages/migration-waves/components/manage-applications-form.tsx

```diff
@@ -216,12 +217,20 @@ export const ManageApplicationsForm: React.FC<ManageApplicationsFormProps> = ({
     onClose();
   };
 
+  console.log(migrationWave);
+
   return (
     <Form onSubmit={onSubmit}>
       <TextContent>
         <Text component={TextVariants.h5}>Selected wave</Text>
         <TextInput
-          value={migrationWave.name}
+          value={
+            migrationWave?.name === ""
```

Comment:
Are we sure `migrationWave.name` will be `""` here or could it be undefined? Elsewhere you check if it is `null`. I wonder if to be safe any place that needs to fall back if the name is missing should work for `""`, `null` or `undefined` values, so it could just check if the name is falsy.

```
!migrationWave?.name ? ... : ...
```

Reply:
@mturley, yes the backend always has `migrationWave.name`  field defined where it could be empty (""). 
Your approach is slicker.